src/libYARP_dev/src/devices/RobotDescriptionServer/RobotDescriptionServer.cpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2016 iCub Facility - Istituto Italiano di Tecnologia
+ * Author: Marco Randazzo <marco.randazzo@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <yarp/dev/IRobotDescription.h>
+#include "RobotDescriptionServer.h"
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/LockGuard.h>
+
+/*! \file Navigation2DClient.cpp */
+
+using namespace yarp::dev;
+using namespace yarp::os;
+using namespace yarp::sig;
+
+
+//------------------------------------------------------------------------------------------------------------------------------
+
+bool yarp::dev::RobotDescriptionServer::open(yarp::os::Searchable &config)
+{
+    m_local_name.clear();
+    m_local_name = config.find("local").asString().c_str();
+
+    if (m_local_name == "")
+    {
+        yError("RobotDescriptionServer::open() error you have to provide valid local name");
+        return false;
+    }
+
+    ConstString local_rpc = m_local_name;
+    local_rpc += "/rpc";
+
+    if (!m_rpc_port.open(local_rpc.c_str()))
+    {
+        yError("RobotDescriptionServer::open() error could not open rpc port %s, check network", local_rpc.c_str());
+        return false;
+    }
+
+    m_rpc_port.setReader(*this);
+    return true;
+}
+
+bool yarp::dev::RobotDescriptionServer::attachAll(const PolyDriverList &p)
+{
+    for (int i = 0; i < p.size(); i++)
+    {
+        //yDebug() << "***************" << p[i]->poly->getOptions().toString();
+        //yDebug() << "***************" << p[i]->poly->getValue("device").toString();
+        //yDebug() << "***************" << p[i]->poly->getValue("name").toString();
+        RobotDescription dev;
+        dev.device_name = p[i]->poly->getValue("name").toString();
```

Comment:
Well, in my opinion it is the other way around, i.e. chosing a fixed param _is_ a problem while having the chance of "mapping" the name _is_ the solution.

This is both for implementation and maintenance reasons.
The parameter `name` can be used by a plugin for whatever reason, for example, if I have a device/plugin which is looking for a file, and I use the keyword `name` for the filename, then my device will be identified with the file name which is not what we want.
The wrappers now conventionally uses the keyword `name` for the port name because, over the years, I modified them to converge on using similar set of parameters in order to have a more uniform look and easy the usage for the user, but originally it was not like this, and we cannot rely on the fact that it will always be.
For devices which use a subdevice, I cannot set a name for device and a different one for the subdevice because I cannot set two `name` parameter.
If we decide the keyword will be `astrupniveral` because this is unlikely to be used by any devices and will be unique, then we need to update all config files of all robots, which is a kind of pain.

Viceversa, implementing a interface means writing 2 lines of code, and you get both advantages of 
- no config file needs to be changed
- sure to avoid name clashes  
- no policy enforced on params to the user
- each device can compose the name the way it prefers. (name + something, or use a specific keyword or whatever. It is a degree of freedom)
- piece of code strongly reliable

If we are mainly interested in the `controlboardwrapper` only 1 device needs to be updated and we get the feature in both robot and gazebo whitout touching anything else.

I see this implementation as a quick hack, not a simple solution because it has pitfalls. I agree with KISS, but is far more important to me that the solution is correct and it will correctly work without any possible problem.

We all know that maintain the config files for the robots is a pain and this is adding more entropy.

_**The more I think about it, the more I'm convinced the current implementation is not the right way to go.**_ 

As said, current implementation could be an educated guess to be used as a fallback for devices which do not yet implement the interface, or as a migration behaviour from now untill all devices have the interface implemented, but I strongly disagree to use it as the final solution.

When me and @randaz81 initially spoke about this interface we do agreed to use the interface for this operation, so I'm a bit surprised of this implementation.